### DISCUSSION TITLE
Authentication Architecture

### DISCUSSION THREAD

**Discussion by sawrubgupta** [lnbits/lnbits#2773] [General]
Hello, I was going through lnbits authentication architecture and was wondering why we used invoice and admin key, not api key and secret key combination at first place which is more safe and widely used.
1. This lacks in access control at more granular level
2. Typically, API keys are paired with secret keys that help to authenticate requests securely. Omitting secret keys can expose your system if the API keys are intercepted, as there would be no second factor to validate the authenticity of the request.
3. Invoice keys, if not adequately secured, could allow unauthorized access to sensitive financial data, which can be a significant risk in payment systems.
4. Without secret key, you lack a secure way to validate webhook signatures
5. Server-to-server communications become more vulnerable to interception.

Please enlighten me 🙏, Its becoming hard for me to understand this type of structure .

**Comment by motorina0** [comment]
This has been addressed.
x-api-key only used for payments now.
